manifests/install.pp

```diff
@@ -68,33 +74,69 @@
       }
     }
   } else {
+    include '::archive'
+
+    $basefilename = "zookeeper-${ensure}.tar.gz"
+    $package_url = "${mirror_url}/zookeeper/zookeeper-${ensure}/${basefilename}"
+    $checksum_url = "http://www-us.apache.org/dist/zookeeper/zookeeper-${ensure}/${basefilename}.sha1"
```

Comment:
Perhaps the checksum should be passed in as an argument to this module. Otherwise an attacker poisoning the tarball merely has to also poison the checksum file and you'd never know if you downloaded a bad version.